subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/InstantExecutionCache.kt

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.instantexecution
+
+import org.gradle.cache.CacheBuilder
+import org.gradle.cache.CacheRepository
+import org.gradle.cache.FileLockManager
+import org.gradle.cache.PersistentCache
+import org.gradle.cache.internal.CleanupActionFactory
+import org.gradle.cache.internal.LeastRecentlyUsedCacheCleanup
+import org.gradle.cache.internal.SingleDepthFilesFinder
+import org.gradle.cache.internal.filelock.LockOptionsBuilder
+import org.gradle.instantexecution.extensions.unsafeLazy
+import org.gradle.instantexecution.initialization.InstantExecutionStartParameter
+import org.gradle.internal.Factory
+import org.gradle.internal.concurrent.Stoppable
+import org.gradle.internal.file.FileAccessTimeJournal
+import org.gradle.internal.resource.local.SingleDepthFileAccessTracker
+import java.io.File
+
+
+internal
+class InstantExecutionCache(
+    private val cacheRepository: CacheRepository,
+    private val cacheCleanupFactory: CleanupActionFactory,
+    private val fileAccessTimeJournal: FileAccessTimeJournal,
+    private val startParameter: InstantExecutionStartParameter
+) : Stoppable {
+
+    private
+    val cacheRootDir
+        get() = startParameter.rootDirectory.resolve(".gradle/configuration-cache")
+
+    private
+    val cleanupDepth = 1
+
+    private
+    val cleanupMaxAgeDays = LeastRecentlyUsedCacheCleanup.DEFAULT_MAX_AGE_IN_DAYS_FOR_RECREATABLE_CACHE_ENTRIES
+
+    private
+    val cache by unsafeLazy {
+        cacheRepository.configurationCache()
+            .withOnDemandLockMode() // Don't need to lock anything until we use the caches
+            .withLruCacheCleanup()
+            .open()
+    }
+
+    private
+    fun CacheRepository.configurationCache() =
+        cache(cacheRootDir).withDisplayName("Configuration Cache")
+
+    private
+    fun CacheBuilder.withOnDemandLockMode() =
+        withLockOptions(LockOptionsBuilder.mode(FileLockManager.LockMode.OnDemand))
+
+    private
+    fun CacheBuilder.withLruCacheCleanup(): CacheBuilder =
+        withCleanup(cacheCleanupFactory.create(LeastRecentlyUsedCacheCleanup(
```

Comment:
:x: I find it a bit hard to read this layout with all parenthesis in a single line. Can we use the usual nested layout here?

Reply:
Reads better?

Reply:
Yes, it does, thank you!